### ISSUE TITLE
DNS via Linux Kernel UID blocked in Android 9

### ISSUE THREAD

**Issue by NeroProtagonist** [ukanth/afwall#867]
Phone: Pixel 2 XL.
Root: Magisk 17.1

I've always had my profile set to Allow Selected and then whitelisted various apps, including Apps running as root (UID 0).
After upgrading to Android 9, this no longer works though. Every DNS request (except for Chrome browser) is filtered, with the log listing occuring under the "Apps running as root(0)'. This effectively prevents internet access.
If I reverse the defaults though by selecting "Block selected" and inverting the selected apps, then DNS will work fine again.

I have tried the NetD options (auto, enable, disable) but they don't seem to change anything.
Attached ipv4rules when DNS doens't work. Weird that no UID 0 DNS is being matched.

How can I diagnose this further?

[IPv4rules.log](https://github.com/ukanth/afwall/files/2412942/IPv4rules.log)

**Comment by ukanth** [comment]
Seems like dns is blocked by default directly in OUTPUT chain.

Can you try this in custom rules with whitelist ?

`$IPTABLES -A "afwall" -p udp --dport 53 -j ACCEPT`

**Comment by NeroProtagonist** [comment]
That fixes the problem, thanks. 

Any idea why this is suddenly a problem? Perhaps DNS was being routed through some Google service before that happened to be whitelisted?
There have been changes in Android 9 with DNS settings with the addition of private DNS mode, where one can specify a custom server in Wifi properties.

**Comment by ukanth** [comment]
Duplicate of #850
